internal/prefixcollector/collector.go

```diff
@@ -21,115 +21,62 @@ import (
 	"cmd-exclude-prefixes-k8s/internal/utils"
 	"context"
 
-	"github.com/networkservicemesh/sdk/pkg/tools/spanhelper"
-
-	apiV1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/watch"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
-
-	"github.com/ghodss/yaml"
-
 	"github.com/sirupsen/logrus"
 
 	"github.com/networkservicemesh/sdk/pkg/tools/prefixpool"
 )
 
-const (
-	configMapKey = "excluded_prefixes.yaml"
-)
-
-// ExcludePrefixSource is source of excluded prefixes
-type ExcludePrefixSource interface {
+// PrefixSource is source of excluded prefixes
+type PrefixSource interface {
 	Prefixes() []string
 }
 
-// ExcludePrefixCollector is service, collecting excluded prefixes from list of ExcludePrefixSource
+// PrefixesWriter is excluded prefixes writer
+type PrefixesWriter interface {
+	Write(context.Context, []string)
+	WatchExcludedPrefixes(context.Context, *utils.SynchronizedPrefixesContainer)
+}
```

Comment:
Can we just pass a function into the collector that means how to write sources?
```suggestion
```

```go
type WritePrefixesFunc func(Config, []strings)
```

Reply:
Replaced writers with WritePrefixesFunc